test: routes for type site and post site

andriacap opened the pull request:
Tests route: 

- [GET] types/site/label
- [POST] site

andriacap committed:
test: routes for type site and post site

Reviewed-by: andriac

mvergez commented on backend/gn_module_monitoring/tests/fixtures/type_site.py:
```diff
@@ -1,10 +1,22 @@
+    jsonfile = os.path.join(folder, filename)
+    with open(jsonfile) as file:
+        data = json.load(file)
+    return data
```
J'utiliserais plutôt la librairie Pathlib qui est plus claire à lire je trouve

andriacap replied:
Pour ça j'ai laissé la librairie os de base juste par soucis de temps à allouer aux modifs

mvergez commented on backend/gn_module_monitoring/tests/test_routes/test_site.py:
```diff
@@ -135,3 +136,37 @@ def test_get_module_sites(self):
+            "sort_label": "label_fr",
+            "sort_dir": "asc",
+        }
+        query_string["label_fr"] = string_contains
```
Possibilité d'inclure directement `label_fr` dans la définition du `query_string`

mvergez commented on backend/gn_module_monitoring/tests/test_routes/test_site.py:
```diff
@@ -135,3 +136,37 @@ def test_get_module_sites(self):
+        assert all([type in r.json["items"] for type in mock_db_type_site])
+
+        query_string["label_fr"] = string_missing
+        r = self.client.get(
```
Il serait mieux de créer 2 tests séparés mais bon ça ferait des répétitions de code...

andriacap replied:
Du coup j'ai laissé un seul test pour la raison de duplication de code

mvergez commented on backend/gn_module_monitoring/tests/fixtures/site.py:
```diff
@@ -42,3 +42,51 @@ def sites(users, types_site, site_group_with_sites):
+        id_nomenclature_type_site=list_nomenclature_id[0],
+        types_site=list_nomenclature_id,
+        id_sites_group=site_group_without_sites.id_sites_group,
+    )
```
Il n'y a pas déjà des sites de créé dans les fixture ? Pas moyen de les réutiliser ?

andriacap replied:
J'ai regardé les sites existant et en fait la logique du test avec deux types de sites et la construction du l'objet  "post" pour la création du nouveau site demanderait à déplacer du code dans la fixture sites . Du coup pas ouf , la création du nouveau site avec deux types de site dans site_to_post_with_types est approprié pour le test "test_post_sites" et ne sert qu'à ce test

mvergez reviewed (changes requested):
Il faudrait passer `black` sur tous les fichiers

andriacap committed:
test: apply black and change query string

- Keep site beauce it's specific one cannot re use existing
- Fix mnemonique "None" with existing label_default

Reviewed-by:andriac
